cmd/flightctl-userinfo-proxy/main.go

```diff
@@ -120,30 +122,62 @@ func makeUserInfoHandler(config *Config, log *logrus.Logger) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		// Only allow GET requests
 		if r.Method != http.MethodGet {
-			http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+			transport.SetResponse(
+				w,
+				nil,
+				api.NewFailureStatus(
+					int32(http.StatusMethodNotAllowed),
+					http.StatusText(http.StatusMethodNotAllowed),
+					"Method not allowed",
+				),
+			)
 			return
 		}
 
 		// Extract Authorization header
 		authHeader := r.Header.Get("Authorization")
 		if authHeader == "" {
-			http.Error(w, "Authorization header required", http.StatusUnauthorized)
+			transport.SetResponse(
+				w,
+				nil,
+				api.NewFailureStatus(
+					int32(http.StatusUnauthorized),
+					http.StatusText(http.StatusUnauthorized),
+					"Authorization header required",
+				),
+			)
 			return
 		}
 
 		// Proxy request to upstream AAP
 		aapResp, err := proxyToUpstream(config, authHeader)
 		if err != nil {
 			log.Printf("Error proxying to upstream AAP: %v", err)
-			http.Error(w, "Internal server error", http.StatusInternalServerError)
+			transport.SetResponse(
+				w,
+				nil,
+				api.NewFailureStatus(
+					int32(http.StatusInternalServerError),
+					http.StatusText(http.StatusInternalServerError),
+					"Internal server error",
```

Comment:
Is the intention to also provide more information? This is just going to come out as {"500", "Internal Server Error", "Internal server error"}, which isn't particularly useful

Reply:
I think your reasoning/decision is solid.